docs/design/resources.md

```diff
@@ -0,0 +1,96 @@
+# Operator Resources
+
+The etcd cluster operator extends the Kubernetes API with two new
+resource definitions:
+
+* `EtcdCluster`, or the "cluster resource", defines an entire
+  operating cluster.
+* `EtcdPeer`, or the "peer resource", defines a single peer of the
+  etcd cluster. This principally concerns a Pod, but may also involve
+  attached storage and network identity.
+
+## The Peer Resource
+
+The peer resource represents a single peer in an etcd
+cluster. Encoding it's bootstrap instructions, persistence settings,
+and extra pod settings (e.g., overridden CPU limits). Its `status`
+field exposes information about the peer including its hostname,
+liveness status, the peer's current generation, and if it's currently
+the leader of the etcd cluster.
+
+When reconciling the etcd peer resource the operator will do the
+following:
+
+1. If storage is enabled and a correctly named persistent volume claim
+   (PVC) does not exist, create one with the settings provided on the
+   peer resource.
+
+   We never delete a persistent volume claim automatically, so this
+   behaviour ensures that the operator will resume stopped clusters if
+   the storage remains even if the peer resource was deleted.
```

Comment:
May be worth adding something like "in future, automatic cluster recovery tools MAY delete PVC resources if they determine it is required and safe to do so, although this is outside of the scope of this document"